cypress/e2e/ui/Settings/Application-Settings/tenant.cy.js

```diff
@@ -0,0 +1,547 @@
+    describe('Validate Edit parent tenant', () => {
+      beforeEach(() => {
+        // Open the tenant edit form
+        cy.toolbar(configToolbarButton, editTenantConfigOption);
+      });
+
+      it('Validate Cancel & Reset buttons on Edit tenant form', () => {
+        // Reset operation
+        resetOperation();
+        // Cancel the edit operation
+        cancelOperation();
+      });
+
+      it('Validate Edit function', () => {
+        // Perform the edit operation
+        addOrEditOperation(
+          saveButton,
+          editedTenantNameValue,
+          editedDescriptionValue,
+          flashMessageSaved,
+          true
+        );
+      });
+
+      after(() => {
+        confirmUiNavigation(() => resetParentTenantForm());
+      });
+    });
+
+    describe('Validate Add Project to parent tenant', () => {
+      it('Validate Add Project function', () => {
+        // Add Project to the parent tenant
+        addProjectToTenant();
+      });
+
+      after(() => {
+        confirmUiNavigation(() => deleteAccordionItem(projectNameValue));
+      });
+    });
+
+    describe('Validate Manage Quotas in parent tenant', () => {
+      beforeEach(() => {
+        // Open the Manage Quotas form
+        cy.toolbar(configToolbarButton, manageQuotasConfigOption);
+      });
+
+      it('Validate Cancel & Reset buttons in Manage Quotas form', () => {
+        // Edit the quotas table and reset the quotas
+        quotasResetOperation();
+        // Cancel the Manage Quotas form
+        cancelOperation();
+      });
+
+      it('Validate Manage Quotas function', () => {
+        // Enabling the quota: "Allocated Storage in GB" with value 10
+        updateQuotas();
+      });
+
+      after(() => {
+        // Resetting the quotas back to all disabled state
+        confirmUiNavigation(() => rollbackQuotas());
+      });
+    });
+  });
+
+  describe('Validate Child Tenant operations: Add, Edit, Add Project, Manage Quotas', () => {
+    describe('Validate Add child tenant function', () => {
+      beforeEach(() => {
+        // Open the Add child tenant form
+        cy.toolbar(configToolbarButton, addChildTenantConfigOption);
+      });
+
+      it('Validate cancel button on Add child tenant form', () => {
+        // Cancel the add operation
+        cancelOperation();
+      });
+
+      it('Validate Add & Delete child tenant', () => {
+        // Adding a child tenant
+        addOrEditOperation();
+        // Delete the created child tenant
+        deleteAccordionItem(initialChildTenantName);
+      });
+
+      it('Validate Adding a duplicate child tenant is restricted', () => {
+        // Adding a child tenant
+        addOrEditOperation();
+        // Open the Add child tenant form again
+        cy.toolbar(configToolbarButton, addChildTenantConfigOption);
+        // Trying to add a child tenant with the same name
+        cy.getFormInputFieldById(nameInputFieldId).type(initialChildTenantName);
```

Comment:
Same as above, passing initialChildTenantName to add on 434 would make this more obvious.